### ISSUE TITLE
[Bug?]: creating new endpoint does not update the deployment serverless (aws) configuration even when using --first-run

### ISSUE THREAD

**Issue by mc-spieczynski** [redwoodjs/redwood#6728]
### What's not working?

TL;DR: if I add a new endpoint to existing app with serverless (aws) configuration present even when running deploy with `yarn rw deploy aws --first-run` parameter will not update the `serverless.yml`.

Also server complained about CORS not about missing endpoint and missing lambda deployment was the real cause of the issue. There are also some other minor issues with documentation/deployment steps mentioned at the end of reproduction steps. Not sure if this is our issue / misconfiguration or just a bug in RW.

### How do we reproduce the bug?

1, Create a RedwoodJS app and deploy to AWS using serverless framework. 
2. Add authentication (dbAuth) that adds custom endpoint (/auth) with separate function (lambda) - how can this be done manually I do not know - please tell me if it's documented, we could test that as well).
3. Deploy the app.

At this point `/auth` is not deployed and not even packaged from RWJS (but I could be wrong about that)

Just for testing we deleted everything from AWS and deployed again with `--first-run`, it will:

4. Run the deployment script with `--first-run` parameter ex. `yarn rw deploy aws --first-run`
5. The zip with auth functions will be created (this should not require running with `--first-run` param but not uploaded as `serverless.yml` configuration should now be manually updated.
6. We reused the `/graphql` config prepared but it is not clear and at least mentioning it in the docs would help or better a console notification that configuration has changed and serverless.yml needs to be updated would be nice. Best if the file was changed (even overwritten) so we can restore changes from VCS and have all things configured properly.

In our case it was also required to configure all paths - to API, to GraphQL and dbAuth even though they are on the same server.
If we did not have GraphQL API path configured it used WEB path for it and for dbAuth we also added configuration.

Moreover I suppose this will be in the serverless docs but we had to configure security groups etc. for AWS - I know it's specific to that framework but knowing nothing about deployments to AWS it was daunting and unclear why things do not work even when CORS was configured properly.

### What's your environment? (If it applies)

```shell
System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 16.13.1 - C:\Users\SEBAST~1\AppData\Local\Temp\xfs-078c7d15\node.CMD
    Yarn: 3.2.1 - C:\Users\SEBAST~1\AppData\Local\Temp\xfs-078c7d15\yarn.CMD
  Browsers:
    Chrome: 106.0.5249.119
    Edge: Spartan (44.19041.1266.0), Chromium (106.0.1370.47)
  npmPackages:
    @redwoodjs/core: ^3.1.1 => 3.1.1
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by dthyresson** [comment]
@joconor Since I know that you user serverless deploys, we think this issue concerns:

1. You setup serverless deploy
2. The config yaml is created with certain functions to deploy
3. You create a new function
4. The yaml isn't updated ... so one needs to manually add any new function to the config so that it gets deployed

If so, we wondered what your workflow is when adding new functions to your app and if you simply manually add them or have some process to automate the creation of the servers config with latest settings?

**Comment by joconor** [comment]
Hey @dthyresson , yeah I just edited the api/serverless.yml manually to add the new function. It can be as simple as just duplicating the `graphql` function block in the YAML and renaming graphql to whatever your function name / handler / endpoint is. I was pleasantly surprised that it was as simple as it is.
Personally, I don't consider automatic generation of additional function blocks in api/serverless.yml to be a bug. It feels like a design choice, although I could see that choice going either way.
If you were to choose to add this as a feature (automatically adding new functions to api/serverless.yml), I think the 'right' place to do that would be in the new function generator, not the `--first-run` serverless deploy. Additionally, I think you'd want `setup auth dbAuth` to do it as well, since (I think) it creates the `auth.js` function. Of course, this would be conditioned on the existence of api/serverless.yml.
In the short term, maybe I could contribute something to the serverless deploy documentation to describe the manual process.

**Comment by dthyresson** [comment]
@mc-spieczynski Hi. Can you review what @joconor suggests and if that works for you?

Also, there is some documentation about "subsequent deploys"

![image](https://user-images.githubusercontent.com/1051633/201194330-fccbecc1-5cb1-4b59-9e8f-e25fd07733ad.png)

Do you think we should add more to the documentation to suggest how to manage the config?

**Comment by mc-spieczynski** [comment]
@dthyresson Hi yeah it does work for me now that I know. 

The "problem" I have with RedwoodJS and any framework really is that there is a lot of ("oh btw..") in documentation (think small print) that needs to be known to work efficiently.

the info from the screenshot also shows that "By default we only configure the `auth` and `graphql` functions for you." - this does not happen if you have run `--first-run` before adding `dbAuth` to the stack. 

So I'd say to at least add `Please run deploy aws --first-run` if you see the serverless files (or any simillar deployment files) AFTER configuring `dbAuth` as auth endpoint will NOT be added otherwise.

I hope this is clear and repeats what @joconor said: "Additionally, I think you'd want setup auth dbAuth to do it as well, since (I think) it creates the auth.js function. Of course, this would be conditioned on the existence of api/serverless.yml."
